### PR TITLE
Switch to f-strings in the PSyIR source

### PR DESCRIPTION
**Pull request by nmnobre**

### CODE REVIEW

File: src/psyclone/psyir/transformations/omp_taskwait_trans.py [hunk 1]
```diff
@@ -164,19 +164,17 @@ def validate(self, node, options=None):
...
-                                                   "dependencies due to "
-                                                   "taskloop dependencies "
-                                                   "across barrierless OMP "
-                                                   "serial regions. Dependency"
-                                                   " is from\n{0}\nto\n{1}"
-                                           .format(
-                                               fwriter(taskloop).rstrip("\n"),
-                                               fwriter(forward_dep)
-                                                   .rstrip("\n"))))
+                    raise TransformationError(LazyString(
```
Comment (arporter) [inline src/psyclone/psyir/transformations/omp_taskwait_trans.py:169]: Please add a comment here explaining why you have `chr(10)`. Could you have `'\n'` instead?

File: src/psyclone/psyir/transformations/omp_taskwait_trans.py [hunk 2]
```diff
@@ -253,9 +251,9 @@ def get_forward_dependence(taskloop, root):
         if parent_parallel is None:
             fwriter = FortranWriter()
             raise InternalError(
-                    LazyString(lambda: "No parent parallel directive was "
-                                       "found for the taskloop region: {0}"
-                               .format(fwriter(taskloop).rstrip("\n"))))
+                    LazyString(lambda: f"No parent parallel directive was "
+                                       f"found for the taskloop region: "
+                                       f"{fwriter(taskloop).rstrip(chr(10))}"))
```
Comment (arporter) [inline src/psyclone/psyir/transformations/omp_taskwait_trans.py:256]: Again, comment about `chr(10)` please (if it's not possible/wise to reinstate the `'\n'`?).

**Review by arporter — CHANGES_REQUESTED** [review]
Good stuff Nuno. In general makes the code more readable I think.
There are a few places where you've touched lines that use the old `six.raise_from` so I'm requesting that you update those. I also have a query about your use of `chr(10)` to replace `'\n'`.
All tests still pass with compilation enabled and the examples are OK too.
Build of ref. guide is unchanged.

Just a note to remind you to let me close the conversations as there's a lot to keep track of :-)

**Comment by nmnobre** [comment]
Hi @arporter, I believe the latest commit addresses all the issues you raised. 🙂

File: src/psyclone/psyir/transformations/omp_taskwait_trans.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/psyir/transformations/omp_taskwait_trans.py:170]: Great, thanks.

**Review by arporter — CHANGES_REQUESTED** [review]
Almost there now. I would have merged it but we've gained a lot of pycodestyle errors:

    $ pycodestyle psyclone/
    psyclone/psyir/transformations/arrayrange2loop_trans.py:141:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/transformations/arrayrange2loop_trans.py:289:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/transformations/fold_conditional_return_expressions_trans.py:101:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/transformations/omp_taskwait_trans.py:168:78: W291 trailing whitespace
    psyclone/psyir/transformations/omp_taskwait_trans.py:253:70: W291 trailing whitespace
    psyclone/psyir/frontend/fparser2.py:1893:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/nodes/array_of_structures_mixin.py:104:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/nodes/omp_directives.py:891:25: E128 continuation line under-indented for visual indent
    psyclone/psyir/symbols/datasymbol.py:232:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/symbols/symbol.py:190:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/backend/language_writer.py:235:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/backend/opencl.py:226:80: E501 line too long (80 > 79 characters)
    psyclone/psyir/backend/sir.py:261:19: E128 continuation line under-indented for visual indent
    psyclone/psyir/backend/sir.py:262:18: E128 continuation line under-indented for visual indent
    psyclone/psyir/backend/sir.py:263:18: E128 continuation line under-indented for visual indent
    psyclone/psyir/backend/sir.py:264:18: E128 continuation line under-indented for visual indent
    psyclone/psyir/backend/sir.py:265:18: E128 continuation line under-indented for visual indent
    psyclone/psyir/backend/sir.py:266:18: E128 continuation line under-indented for visual indent

Apart from that, tests and examples are all OK with compilation.
Ref guide still builds fine.

**Comment by nmnobre** [comment]
> Almost there now. I would have merged it but we've gained a lot of pycodestyle errors:
> 
> ```
> $ pycodestyle psyclone/
> psyclone/psyir/transformations/arrayrange2loop_trans.py:141:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/transformations/arrayrange2loop_trans.py:289:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/transformations/fold_conditional_return_expressions_trans.py:101:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/transformations/omp_taskwait_trans.py:168:78: W291 trailing whitespace
> psyclone/psyir/transformations/omp_taskwait_trans.py:253:70: W291 trailing whitespace
> psyclone/psyir/frontend/fparser2.py:1893:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/nodes/array_of_structures_mixin.py:104:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/nodes/omp_directives.py:891:25: E128 continuation line under-indented for visual indent
> psyclone/psyir/symbols/datasymbol.py:232:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/symbols/symbol.py:190:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/backend/language_writer.py:235:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/backend/opencl.py:226:80: E501 line too long (80 > 79 characters)
> psyclone/psyir/backend/sir.py:261:19: E128 continuation line under-indented for visual indent
> psyclone/psyir/backend/sir.py:262:18: E128 continuation line under-indented for visual indent
> psyclone/psyir/backend/sir.py:263:18: E128 continuation line under-indented for visual indent
> psyclone/psyir/backend/sir.py:264:18: E128 continuation line under-indented for visual indent
> psyclone/psyir/backend/sir.py:265:18: E128 continuation line under-indented for visual indent
> psyclone/psyir/backend/sir.py:266:18: E128 continuation line under-indented for visual indent
> ```
> 
> Apart from that, tests and examples are all OK with compilation. Ref guide still builds fine.

Just to clarify, the majority of these were already there before your previous comment right?

**Comment by nmnobre** [comment]
> > Just to clarify, the majority of these were already there before your previous comment right?
> 
> I don't know when they appeared but they're not there on master :-)

Sure, I was just a bit confused since I didn't touch the majority of those files in yesterday's commits.

**Comment by nmnobre** [comment]
Done. 😊

Since I'm here, do you also see a bunch of E302 errors under `./tutorial/practicals/LFRic/distributed_memory` ?

**Comment by arporter** [comment]
Oh yes, I do! I never normally run pycodestyle over that (my bad).

**Comment by nmnobre** [comment]
Sorted. 😄

**Review by arporter — APPROVED** [review]
Code is squeaky clean now.
Tests and examples all still OK after bringing branch up to master.
Will proceed to merge.

**Pull request merged**